fili-core/src/main/java/com/yahoo/bard/webservice/table/resolver/GranularityComparator.java

```diff
@@ -15,6 +15,24 @@
 public class GranularityComparator implements Comparator<PhysicalTable> {
 
     private static final Logger LOG = LoggerFactory.getLogger(GranularityComparator.class);
+    private static GranularityComparator granularityComparator;
+
+    /**
+     * Disable constructor to force using getInstance() method.
+     */
+    private GranularityComparator() { }
+
+    /**
+     * Factory method for creating new GranularityComparator instance.
+     *
+     * @return a new GranularityComparator instance
+     */
+    public static synchronized GranularityComparator getInstance() {
+        if (granularityComparator == null) {
```

Comment:
Remove this condition. We should just instantiate the instance variable once when the class is loaded, rather than incur the overhead of the check every time.